test/adapter.test.ts

```diff
@@ -25,164 +22,186 @@ const MIN_ADA_POOL_V1_ID_TESTNET =
+  blockfrostAdapterTestnet = new BlockfrostAdapter(
     NetworkId.TESTNET,
     new BlockFrostAPI({
       projectId: mustGetEnv("BLOCKFROST_PROJECT_ID_TESTNET"),
       network: "preprod",
     })
   );
-  adapterMainnet = new BlockfrostAdapter(
+  blockfrostAdapterMainnet = new BlockfrostAdapter(
     NetworkId.MAINNET,
     new BlockFrostAPI({
       projectId: mustGetEnv("BLOCKFROST_PROJECT_ID_MAINNET"),
       network: "mainnet",
     })
   );
+  const cardanoNetworkPreprod: Network = "Preprod";
+  maestroAdapterTestnet = new MaestroAdapter(
+    NetworkId.TESTNET,
+    new MaestroClient(
+      new Configuration({
+        apiKey: "MAESTRO_API_KEY_TESTNET",
+        network: cardanoNetworkPreprod,
+      })
+    )
+  );
+  const cardanoNetworkMainnet: Network = "Mainnet";
+  maestroAdapterMainnet = new MaestroAdapter(
+    NetworkId.MAINNET,
+    new MaestroClient(
+      new Configuration({
+        apiKey: "MAESTRO_API_KEY_MAINNET",
+        network: cardanoNetworkMainnet,
+      })
+    )
+  );
 });
 
-test("getAssetDecimals", async () => {
-  expect(await adapterTestnet.getAssetDecimals("lovelace")).toBe(6);
-  expect(await adapterTestnet.getAssetDecimals(MIN_TESTNET)).toBe(0);
-  expect(await adapterMainnet.getAssetDecimals("lovelace")).toBe(6);
-  expect(await adapterMainnet.getAssetDecimals(MIN_MAINNET)).toBe(6);
-});
-
-async function testPoolPrice(adapter: BlockfrostAdapter): Promise<void> {
-  const pools = await adapter.getV1Pools({
-    page: 1,
+async function runTests(adapterTestnet: any, adapterMainnet: any) {
+  test("getAssetDecimals", async () => {
+    expect(await adapterTestnet.getAssetDecimals("lovelace")).toBe(6);
+    expect(await adapterTestnet.getAssetDecimals(MIN_TESTNET)).toBe(0);
+    expect(await adapterMainnet.getAssetDecimals("lovelace")).toBe(6);
+    expect(await adapterMainnet.getAssetDecimals(MIN_MAINNET)).toBe(6);
   });
-  expect(pools.length).toBeGreaterThan(0);
-  // check random 5 pools
-  for (let i = 0; i < 5; i++) {
-    const idx = Math.floor(Math.random() * pools.length);
-    const pool = pools[idx];
-    const [priceAB, priceBA] = await adapter.getV1PoolPrice({ pool });
-    // product of 2 prices must be approximately equal to 1
-    // abs(priceAB * priceBA - 1) <= epsilon
-    expect(priceAB.mul(priceBA).sub(1).abs().toNumber()).toBeLessThanOrEqual(
-      1e-6
-    );
-  }
-}
 
-test("getPoolPrice", async () => {
-  await testPoolPrice(adapterTestnet);
-  await testPoolPrice(adapterMainnet);
-}, 10000);
+  async function testPoolPrice(adapter: any): Promise<void> {
+    const pools = await adapter.getV1Pools({ page: 1 });
+    expect(pools.length).toBeGreaterThan(0);
+    for (let i = 0; i < 5; i++) {
+      const idx = Math.floor(Math.random() * pools.length);
+      const pool = pools[idx];
+      const [priceAB, priceBA] = await adapter.getV1PoolPrice({ pool });
+      expect(priceAB.mul(priceBA).sub(1).abs().toNumber()).toBeLessThanOrEqual(
+        1e-6
+      );
+    }
+  }
```

Comment:
can you keep my comments here

Reply:
what are you referring to?

Reply:
@Vardominator in the old code I have code comments to explain the formula :p (the `// blabla` parts)